helm/prometheus-rules/templates/platform/atlas/alerting-rules/promtail.rules.yml

```diff
@@ -13,7 +13,18 @@ spec:
           annotations:
             description: '{{`Scraping of all promtail pods to check if one failed every 30 minutes.`}}'
             opsrecipe: promtail/
-          expr: count(up{container="promtail"} == 0) by (cluster_id, installation, provider, pipeline) > 0
+          expr: |-
+            (
+              # List promtail pods to be able to get the node label and join with the node status to not alert if the node is not ready
```

Comment:
I feel like having an other alert/inhibition for node not ready would make this expression easier. WDYT ?

Reply:
It would replace the last section yes:

```
* on (node) group_left()
              (
                kube_node_status_condition{condition="Ready", status="true"} == 1
              )
```

But I'm really keen on using inhibition without a paging alert behind it

Reply:
An inhibition would be better to:
* make the alert easier to read
* make the "check if node is ready" part easy to share for other alerts, because promtail is probably not the only alert requiring this inhibition

However, the caveats of the inhibition are:
* you must ensure the inhibition alert fires before the inhibited alert
* you must ensure the inhibition alert keeps firing longer than the inhibited alert
* even when this timing is good, sometimes alerts get through the inhibition

So you may still get false positives.

Also, even if we use an inhibition the join with `node` is still required.


If we plan to do something similar for some other alerts (ie inhibit on node not ready), I prefer the inhibition.
If this is the only alert requiring this, I don't mind having the condition hardcoded in the alert query.